src/magmaviz/scatterplot.py

```diff
@@ -21,5 +54,144 @@ def scatterplot(df, x, y, c=""):
+        raise TypeError("Invalid value passed to 'ytitle' variable: Assign y-axis title as a 'string'.")
+
+    # check if color legend title is a string
+    if not isinstance(ctitle, str):
+        raise TypeError("Invalid value passed to 'ctitle' variable: Assign color legend title as a 'string'.")
+
+    # check if xzero is a boolean
+    if not isinstance(xzero, bool):
+        raise TypeError("Invalid value passed to 'xzero' variable: Assign boolean True to begin x axis from zero.")
+
+    # check if yzero is a boolean
+    if not isinstance(yzero, bool):
+        raise TypeError("Invalid value passed to 'ctitle' variable: Assign boolean True to begin y axis from zero.")
+
+    # check if shapes is a boolean
+    if not isinstance(shapes, bool):
+        raise TypeError("Invalid value passed to 'shapes' variable: Assign boolean True to show different shapes for each color category.")
+
+    # Advanced checks to see whether columns exists, opacity and size are within expected range, etc.
+    # check if column name assigned to x-axis is present in the dataframe
+    assert x in list(
+        df.columns
+    ), "The column specified for 'x' axis does not exist in the dataframe."
+
+    # check if column name assigned to y-axis is present in the dataframe
+    assert y in list(
+        df.columns
+    ), "The column specified for 'y' axis does not exist in the dataframe."
+
+    # check if column name assigned to color is present in the dataframe
+    if c != "":
+        assert c in list(
+            df.columns
+        ), "The column specified for 'color' does not exist in the dataframe."
+
+    # check if opacity value is in the range 0.1 to 1.0
+    if o <= 0.1 or o > 1.0:
+        raise TypeError("Opacity value must be in the range [0.1, 1.0]")
+
+    # check if size value is in the range 1 to 100
+    if s < 1 or s > 100:
+        raise TypeError("Size of data points must be in the range [1, 100]")
+
+    # check if x-axis column is numeric or not
+    assert is_numeric_dtype(df[x]), "The column assigned to 'x' axis is not of type numeric."
+
+    # check if y-axis column is numeric or not
+    assert is_numeric_dtype(df[y]), "The column assigned to 'y' axis is not of type numeric."
+
+    # check if color column is string or not
+    if c != "":
+        assert is_string_dtype(df[c]), "The column assigned to 'color' is not of type string."
+
+    # add proper titles to axes, legend and plot
+    if xtitle == "":
+        xtitle = re.sub(r"[_.,-]", " ", x)
+    if ytitle == "":
+        ytitle = re.sub(r"[_.,-]", " ", y)
+    if ctitle == "":
+        ctitle = re.sub(r"[_.,-]", " ", c)
+    if t == "":
+        if c == "":
+            t = f"{xtitle.title()} vs {ytitle.title()}"
+        else:
+            t = f"{xtitle.title()} vs {ytitle.title()} by {ctitle.title()}"
+
+    # scatterplot
+    if c == "":
+        plot = alt.Chart(
+            data=df, title=t
+        ).mark_point(
+            opacity=o, size=s, color="purple"
```

Comment:
Just curious, would setting the color to purple here override the magma theme?

Reply:
Oh yes, good catch. I wanted to have a placeholder color and later update it based on the magma theme. Will update it soon.

Reply:
Sounds good!